### PR TITLE
test(api): EntitlementsResponse DTO + entitlement/re-sell focus coverage

### PR DESCRIPTION
**Pull request by mablack01**
## Summary

Introduces `EntitlementsResponse` DTO + static mapper, replacing the hand-rolled `JsonObject` walk in `ApiHttpTransport.fetchEntitlementsAsync` with a characterization-pinned identical implementation. Adds 31 new unit tests covering entitlement parsing edge cases, stale re-sell focus tri-state branches, tick-retry guard regressions, and seam-DTO round-trips from the #731 redesign. No production behavior changes — the refactor is fully pinned by the new test suite.

### ♻️ Refactoring
- New `com.flipsmart.api.dto.EntitlementsResponse` DTO with static `fromJson` mapper replaces the inline `JsonObject` walk in `ApiHttpTransport.fetchEntitlementsAsync`; `is_premium` held as a raw `JsonElement` so a non-primitive value safely defaults to `false` (no behavioral change)
- Removed now-unused `JSON_KEY_RSN_ENTITLEMENT` constant from `ApiHttpTransport`
- Added `ApiHttpTransport.parse(body, Class)` helper (behavior-neutral Gson delegate); adopted in pure-DTO endpoint groups (Motd/BankSnapshot/Blocklist/Flips/OfferAction), dropping their redundant `gson` field

### ✅ Tests — +31 unit tests
- `EntitlementsResponseTest` (13 tests): pins every premium branch (`true`/`false`/missing/non-primitive/`null`) and rsn-entitlement branch (blocked/non-blocked/missing-status/null/missing-key) + null/empty-body safety
- `OverrideFocusForSellTest` (5 tests): covers `overrideFocusForSell` tri-state UNAVAILABLE branches (inactive, null session, unresolvable price, zero quantity) and the clean FOCUSED-fires-callback path
- `SellFocusRetryTest` (3 tests): pins `GrandExchangeTracker` tick-retry — settles to FOCUSED within budget without API fallback; falls back to `tryAsyncSellFocus`/`getActiveFlipsAsync` exactly on budget exhaustion (guards the `3c6e53d` regression); clears pending retry when auto-recommend is turned off
- `ApiHttpTransportParseTest` (3 tests): covers the new `parse` helper
- `SeamDtoDeserializationTest` (7 tests): round-trip coverage for the #731 seam response DTOs (`FlipStatistics`/`FlipFinder`/`CompletedFlips`/`Blocklists`/`BankSnapshotStatus`), pinning snake_case wire-key mappings

### Automated Tests
- `./gradlew build` — 324 tests, 0 failures, 0 errors
- Checkstyle: no violations
- All tests UP-TO-DATE after full build

### Manual Testing
- No production behavior change; manual testing not required for this PR

### Acceptance Criteria
- [x] `EntitlementsResponse` DTO + mapper replaces manual `JsonObject` parsing (behavior identical, pinned by test suite)
- [x] Unit tests cover all premium branches (`true`/`false`/missing/non-primitive/`null`) and all rsn-entitlement branches
- [x] `SellFocusRetryTest` guards the `3c6e53d` regression (budget-exhaustion fallback path)
- [x] `./gradlew build` green — 324 tests pass
- [x] No issue-ref comments in plugin main source (LOC budget — verified by diff scan)

## Deployment Notes

- No migration required
- No new dependencies
- No environment variable changes
- Pure test/refactor PR — no behavioral change to ship separately

## Checklist

- [x] Tests added — 31 new unit tests
- [x] No console.log / debug code left
- [x] Source comments clean (no issue refs in `src/main/java/`, test javadoc is exempt)
- [x] Commits are clean and descriptive
- [x] Build passes locally

## Related Issues

Closes Flip-Smart/flip-smart#750

---

## 🤖 Codacy AI Reviewer — 4 comments processed

| Disposition | File | Note |
|---|---|---|
| Fixed (ac3f377) | `EntitlementsResponse.java:20` | Renamed field `isPremium` → `premiumElement` to resolve `PMD_AvoidFieldNameMatchingMethodName` |
| False Positive | `SellFocusRetryTest.java:30` | `PMD_SingularField` does not account for JUnit `@Before` / Mockito fixture lifecycle |
| Fixed (46ea3f5) | `ApiHttpTransport.java:91` | Rephrased incomplete Javadoc sentence for grammatical clarity |
| False Positive | `ApiHttpTransport.java:43` | `JSON_KEY_IS_PREMIUM` (line 507) and `JSON_KEY_STATUS` (line 1011) are actively used; only `JSON_KEY_RSN_ENTITLEMENT` was removed |

## 🩺 Codacy Quality Gate — 6 issues resolved

| Disposition | Count | Rule | Files |
|---|---|---|---|
| Fixed (ac3f377) | 1 | `PMD_AvoidFieldNameMatchingMethodName` | `EntitlementsResponse.java` |
| Dismissed | 5 | `PMD_SingularField` | `SellFocusRetryTest.java` (×4), `OverrideFocusForSellTest.java` (×1) — false positives in JUnit/Mockito test fixture context |

### CODE REVIEW

File: src/main/java/com/flipsmart/api/dto/EntitlementsResponse.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * <p>Premium is held as a raw {@link JsonElement} rather than a {@code Boolean}
+ * so a non-primitive {@code is_premium} (object/array) defaults to false instead
+ * of throwing during deserialization, preserving the original defensive parse.
+ */
+public class EntitlementsResponse
+{
+	private static final String STATUS_BLOCKED = "blocked";
+
+	@SerializedName("is_premium")
+	private JsonElement isPremium;
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/api/dto/EntitlementsResponse.java]: Fixed in ac3f377 — renamed field `isPremium` → `premiumElement` so the field and method no longer share a name. The `@SerializedName("is_premium")` annotation continues to handle JSON mapping.

File: src/test/java/com/flipsmart/SellFocusRetryTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * tracker retries once per game tick for a bounded budget before falling back to
+ * the backend active-flip lookup — the fallback restored in {@code 3c6e53d}.
+ */
+public class SellFocusRetryTest
+{
+	private static final int ITEM = 4151;
+	private static final String RSN = "TestPlayer";
+	private static final int RETRY_BUDGET_TICKS = 6;
+
+	private PlayerSession session;
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/test/java/com/flipsmart/SellFocusRetryTest.java:30]: False positive — these are JUnit test fixture fields managed by the `@Before setUp()` lifecycle. Making them local to `setUp()` would require duplicating the construction chain inside each test method. Keeping them at class level is the standard Mockito/JUnit4 pattern and improves readability; PMD's `SingularField` rule does not account for the framework-driven lifecycle.

File: src/main/java/com/flipsmart/api/ApiHttpTransport.java [hunk 3]
```diff
@@ -86,6 +86,17 @@ public Gson getGson()
 		return gson;
 	}
 
+	/**
+	 * Deserialize a response body into a DTO using the transport's Gson. Centralizes
+	 * the {@code gson.fromJson(body, T.class)} boilerplate the endpoint groups each
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/api/ApiHttpTransport.java]: Fixed in 46ea3f5 — rephrased to "Centralizes the boilerplate that each endpoint group previously hand-rolled" for grammatical clarity.

File: src/main/java/com/flipsmart/api/ApiHttpTransport.java [hunk 4]
```diff
@@ -39,7 +40,6 @@ public class ApiHttpTransport
 	private static final String ACCESS_TOKEN_KEY = "access_token";
 	private static final String REFRESH_TOKEN_KEY = "refresh_token";
 	private static final String JSON_KEY_IS_PREMIUM = "is_premium";
-	private static final String JSON_KEY_RSN_ENTITLEMENT = "rsn_entitlement";
 	private static final String JSON_KEY_STATUS = "status";
```
[the remark this answers is not in the document]
Comment (mablack01) [reply src/main/java/com/flipsmart/api/ApiHttpTransport.java:43]: False positive — both constants are actively used: `JSON_KEY_IS_PREMIUM` is referenced on line 507 inside `processTokenResponse()` (`tokenResponse.has(JSON_KEY_IS_PREMIUM)`) and `JSON_KEY_STATUS` is referenced on line 1011 inside `pollDeviceStatusAsync()` (`json.get(JSON_KEY_STATUS)`). The diff for this PR only removed `JSON_KEY_RSN_ENTITLEMENT`, which was genuinely replaced by the DTO.
